qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -57,9 +58,47 @@ def _wait_for_results(self, timeout: float, wait: float = 5) -> List[Dict[str, D
         """
 
         self.wait_for_final_state(timeout, wait)  # should call self.status()
-
         return [self._job_info[job_id] for job_id in self._job_id.split(",")]
 
+    def _arrange_counts(
+        self, counts: Dict[str, int], circ_meas_bit_indices: List[int], num_clbits: int
+    ) -> Dict[str, int]:
+        """Arranges the classical bit strings from job counts to match classical register.
+
+        Args:
+            counts: The raw counts from a job result.
+            circ_meas_bit_indices: The indices of the measured qubits.
+            num_clbits: The number of classical bits for the corresponding job circuit.
+
+        Returns:
+            A dictionary with the updated counts keys.
+        """
+        arranged_counts = {}
+        for key in counts.keys():
+            updated_key = "0" * num_clbits
+            for counter, index in enumerate(circ_meas_bit_indices):
+                updated_key = updated_key[:index] + key[counter] + updated_key[index + 1 :]
+            arranged_counts[updated_key] = counts[key]
+
+        return arranged_counts
+
+    def _fetch_meas_info(self, index: int) -> int:
+        """Helper method to update measurement indices and get number of classical bits.
+
+        Args:
+            index: The index of the circuit of interest.
+
+        Returns:
+            The number of classical bits in the circuit.
+        """
+        if self._circ_meas_bit_indices == []:
+            self._circ_meas_bit_indices = [
+                qss.compiler_output.measured_qubit_indices(circuit)
+                for circuit in self.compiled_circuits()
+            ]
+
+        return self._backend._num_clbits_in_circ[index]
```

Comment:
nit: this seems to be doing too different things (the returned value is unrelated to the fetched `_circ_meas_bit_indices`). imo this should either be separated into two functions (or, `self._backend._num_clbits_in_circ[index]` could just be used inline in `result()`)

also do we need to make `_circ_meas_bit_indices` an attribute of the job? it seems like it only needs to be computed once in `result()`, in which case this should just return it as a value

Reply:
I originally had everything inline in `result()` but somewhere along the line, I changed it because I didn't want to re-populate `_circ_meas_bit_indices` every time the user called `result()` or something. I've updated the code based on your suggestion now.